### PR TITLE
Feature/30 update a user context

### PR DESCRIPTION
**Pull request by hitohata**
## Overviews of implementation

Updated User and User ID class

## Review points

I deleted the use case test codes but I didn't delete these codes. Because those codes are for the reference of implementation for now.

### CODE REVIEW

**Review by taniguchiiqqq — CHANGES_REQUESTED** [review]
I'm sorry. I'm not sure I could get your point. I just found running the test failed. I agree that leaving some no longer needed test code can be helpful. But if those make the tests fail, those had better be deleted or commented out.

File: backend/main/src/User/Domain/User.ts [hunk 1]
```diff
@@ -3,26 +3,20 @@ import { DomainObjectError } from "src/Shared";
...
+ * The name of user must be less than equal to 30.
+ */
 export class User extends AggregateRoot<string, IUserProps> {
-	// username must be shorter than 100.
-	static create(
-		id: UserId,
-		props: IUserPropsInput,
-	): Result<User, UserDomainError> {
-		if (props.name.length > 100) {
+	static create(id: UserId, props: IUserProps): Result<User, UserDomainError> {
```
Comment (taniguchiiqqq) [inline backend/main/src/User/Domain/User.ts:13]: These changes require the following code to be modified. I'm not sure if it is the best way, but I mean, a type guard seems needed.

> backend/main/src/User/UseCases/UpdateUserUseCase/UpdateUserNameUseCase.ts: 51

```typescript
if (!nameUpdatedUser.ok) {
	return Err(nameUpdatedUser.error);
}

await this.userRepository.update(nameUpdatedUser.value);
```

File: backend/main/src/User/Domain/User.ts [hunk 2]
```diff
@@ -31,20 +25,28 @@ export class User extends AggregateRoot<string, IUserProps> {
 		return this.props.name;
 	}
 
-	get isAdmin(): boolean {
```
Comment (taniguchiiqqq) [inline backend/main/src/User/Domain/User.ts:34]: This change makes codes for checking user types unnecessary.
It indicates the following codes and some test codes.

> backend/main/src/User/UseCases/UpdateUserUseCase/UpdateUserNameUseCase.ts: 44-

File: backend/main/src/User/Domain/User.ts [hunk 3]
```diff
@@ -31,20 +25,28 @@ export class User extends AggregateRoot<string, IUserProps> {
...
+/**
+ * This class is the user's ID.
+ */
 export class UserId extends Identifier<string> {
 	static create(id: string): Result<UserId, UserIdDomainError> {
+		const regId =
+			/^[a-z 0-9]{8}-[a-z 0-9]{4}-[a-z 0-9]{4}-[a-z 0-9]{4}-[a-z 0-9]{12}$/;
+
+		if (!regId.test(id)) {
+			return Err(new UserIdDomainError("Incorrect User ID pattern"));
```
Comment (taniguchiiqqq) [inline backend/main/src/User/Domain/User.ts:47]: Thank you for adding a Domain Logic to the UserId class! I found it causes some errors in the tests. Please check and fix them!

**Comment by hitohata** [comment]
@taniguchiiqqq 
Thank you for your review.
I'm sorry. It seems that I couldn't push the 'delete test' commit correctly.
What I wanted to say is that the use cases and controller were written for mock or reference of implementation of other use cases.
I think it should be rewritten when we write the use cases and controllers but it is not now. Therefore, I Think I should not delete it for now.

**Comment by taniguchiiqqq** [comment]
Thank you for your update. Make sense. I confirmed that it passed the tests.
I appreciate your explanation. I totally agree that those codes should be modified later.

**Review by taniguchiiqqq — APPROVED** [review]

**Pull request merged**